README.md

```diff
@@ -0,0 +1,209 @@
+
+## Project Structure and Functionality
+
+The project is organized into several key components, each responsible for specific functionalities:
+
+### Controllers
+
+- **AuthController**: Manages user-related operations such as registration, login, and profile management.
+- **BookController**: Manages book processing, including book creation and updating for admins, and viewing for users.
+- **OrderController**: Manages order processing, including order creation, updating, and viewing order history.
+- **CategoryController**: Manages category processing, including category creation, updating, and viewing.
+- **ShoppingCartController**: Manages shopping cart processing, including creation, updating, and viewing items (books and other products).
+
+### Services
+
+- **OrderService**: Manages the order processing logic, including placing orders in the database, retrieving order history, and calculating the total price of all items.
+- **BookService**: Manages the book processing logic, including book management for admins and book retrieval for users.
+- **CategoryService**: Manages category processing logic, including category management for admins and category retrieval for users.
+- **ShoppingCartService**: Handles the business logic related to shopping carts, including adding items and updating the cart.
+- **UserService**: Contains business logic for user operations such as registration and user retrieval by username (email).
+
+### Repositories
+
+- **BookRepository**: Interface for CRUD operations on Book entities and special methods for finding books by category.
+- **ShoppingCartRepository**: Interface for CRUD operations on ShoppingCart entities and methods for identifying a shopping cart by user ID.
+- **CartItemRepository**: Interface for CRUD operations on CartItem entities.
+- **OrderRepository**: Interface for CRUD operations on Order entities.
+- **OrderItemRepository**: Interface for CRUD operations on OrderItem entities and methods for retrieving order items by order ID.
+- **UserRepository**: Interface for CRUD operations on User entities and methods for checking if the user is already registered.
+- **RoleRepository**: Interface for CRUD operations on Role entities, including methods for identifying user roles.
+
+## Setup and Usage
+
+To get started with this project, follow these steps:
+
+1. **Clone the repository**:
+    ```sh
+    git clone https://github.com/bodya4243/online-book-store.git
+    cd online-book-store
+    ```
+
+2. **Build the project using Maven**:
+    ```sh
+    mvn clean install
+    ```
+
+3. **Run the application**:
+    ```sh
+    mvn spring-boot:run
+    ```
+   Technologies and Tools
+   Key Technologies
+   Java Development Kit (JDK) 17
+   Maven
+   MySQL
+   Spring Boot
+   H2 Database (for testing)
+   Lombok
+   MapStruct
+   Liquibase
+   SpringDoc OpenAPI
+   JWT (JSON Web Token)
+   Docker Compose
```

Comment:
* duplicated info 
* it is not formatted
* it should not be in the middle of the "running the app" guide